packages/core/src/animation.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {Injectable} from './di';
+
+/**
+ * The event type for when `animate.enter` and `animate.leave` are used with function
+ * callbacks.
+ *
+ * @publicApi 20.2
+ */
+export type AnimationCallbackEvent = {target: Element; animationComplete: Function};
+
+/**
+ * Animation removal functions have a four second long maximum duration timeout.
+ * This value mirrors from Chrome's cross document navigation view transition timeout.
+ * It's intended to prevent people from accidentally forgetting to call the removal
+ * function in their callback.
+ */
+const ANIMATION_TIMEOUT = 4000;
```

Comment:
nit: more descriptive const name:
```suggestion
const MAX_ANIMATION_DURATION = 4000;
```